### PR TITLE
Corrected Redirect Page with Shimmer Effect and Reload Button for Error Message

### PR DESCRIPTION
**Pull request by sahsisunny**
## Issue Ticket Number

-   #86

## Description

In this pull request, the redirect page has been enhanced with the addition of a shimmer effect for a more visually appealing user experience. An error message has also been incorporated, accompanied by a reload button to improve user interaction. The changes have been made to address issue #86.

**Breaking Changes**

-   [ ] Yes
-   [x] No

**Development Tested?**

-   [x] Yes
-   [ ] No

**Tested in Staging?**

-   [x] Yes
-   [ ] No

**Under Feature Flag**

-   [x] Yes
-   [ ] No

**Database Changes**

-   [x] No
-   [ ] Yes

## Test Coverage

File                          | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------------------|---------|----------|---------|---------|-------------------
All files                     |   97.36 |     92.1 |   93.67 |   98.45 |                   
 fixtures                     |     100 |      100 |     100 |     100 |                   
  users.ts                    |     100 |      100 |     100 |     100 |                   
 src/components/App           |     100 |      100 |     100 |     100 |                   
  InputSection.tsx            |     100 |      100 |     100 |     100 |                   
  OutputSection.tsx           |     100 |      100 |     100 |     100 |                   
 src/components/Button        |     100 |      100 |     100 |     100 |                   
  index.tsx                   |     100 |      100 |     100 |     100 | 
 src/components/Dashboard     |     100 |      100 |     100 |     100 | 
  NoUrlFound.tsx              |     100 |      100 |     100 |     100 | 
  UrlList.tsx                 |     100 |      100 |     100 |     100 | 
  UrlListItem.tsx             |     100 |      100 |     100 |     100 | 
 src/components/Footer        |     100 |      100 |     100 |     100 | 
  index.tsx                   |     100 |      100 |     100 |     100 | 
 src/components/Layout        |     100 |      100 |     100 |     100 | 
  index.tsx                   |     100 |      100 |     100 |     100 | 
 src/components/LoginModal    |     100 |      100 |     100 |     100 | 
  index.tsx                   |     100 |      100 |     100 |     100 | 
 src/components/Navbar        |   94.11 |      100 |   71.42 |   93.54 | 
  NavbarMenuItems.tsx         |     100 |      100 |     100 |     100 | 
  UserProfileButton.tsx       |     100 |      100 |     100 |     100 | 
  index.tsx                   |   90.47 |      100 |      50 |      90 | 19,54
 src/components/ProfileIcon   |     100 |      100 |     100 |     100 | 
  ProfileIcon.tsx             |     100 |      100 |     100 |     100 | 
 src/components/Redirect      |     100 |      100 |     100 |     100 | 
  ErrorPage.tsx               |     100 |      100 |     100 |     100 | 
  LoaderTimer.tsx             |     100 |      100 |     100 |     100 | 
  NotFound.tsx                |     100 |      100 |     100 |     100 | 
  RedirectFooter.tsx          |     100 |      100 |     100 |     100 | 
 src/components/ShimmerEffect |     100 |      100 |     100 |     100 | 
  DashboardShimmer.tsx        |     100 |      100 |     100 |     100 | 
  OutputSectionShimmer.tsx    |     100 |      100 |     100 |     100 | 
  RedirectShimmer.tsx         |     100 |      100 |     100 |     100 | 
  UserLoginShimmer.tsx        |     100 |      100 |     100 |     100 | 
 src/components/Toast         |     100 |      100 |     100 |     100 | 
  index.tsx                   |     100 |      100 |     100 |     100 | 
 src/components/icons         |     100 |      100 |     100 |     100 | 
  redirect.tsx                |     100 |      100 |     100 |     100 | 
  signWithGoogle.tsx          |     100 |      100 |     100 |     100 | 
 src/constants                |   89.47 |      100 |     100 |     100 | 
  constants.ts                |   83.33 |      100 |     100 |     100 | 
  url.ts                      |    92.3 |      100 |     100 |     100 | 
 src/hooks                    |   96.42 |    66.66 |    90.9 |     100 | 
  useAuthenticated.ts         |     100 |      100 |     100 |     100 | 
  useToast.ts                 |   95.23 |        0 |      90 |     100 | 10
 src/pages/[redirect]         |   86.11 |       75 |   66.66 |   90.32 | 
  index.tsx                   |   86.11 |       75 |   66.66 |   90.32 | 31-32,38
 src/pages/app                |     100 |      100 |     100 |     100 | 
  index.tsx                   |     100 |      100 |     100 |     100 | 
 src/pages/dashboard          |     100 |      100 |     100 |     100 | 
  index.tsx                   |     100 |      100 |     100 |     100 |                   
 src/services                 |     100 |      100 |     100 |     100 | 
  api.tsx                     |     100 |      100 |     100 |     100 | 
 src/utils                    |     100 |       90 |     100 |     100 | 
  formatDate.ts               |     100 |    88.88 |     100 |     100 | 7
  validateUrl.ts              |     100 |      100 |     100 |     100 | 
------------------------------|---------|----------|---------|---------|-------------------
## Additional Notes

No additional considerations.

### CODE REVIEW

File: src/pages/[redirect]/index.tsx [hunk 1]
```diff
@@ -2,9 +2,10 @@ import Head from 'next/head';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
 
+import ErrorPage from '@/components/Redirect/ErrorPage';
 import LoaderTimer from '@/components/Redirect/LoaderTimer';
-import NotFound from '@/components/Redirect/NotFound';
 import RedirectFooter from '@/components/Redirect/RedirectFooter';
+import RedirectShimmer from '@/components/ShimmerEffect/RedirectShimmer';
 import { useGetOriginalUrlQuery } from '@/services/api';
```
Comment (satyam73) [inline src/pages/[redirect]/index.tsx:9]: try puting this import above the components import statement

File: src/components/ShimmerEffect/RedirectShimmer.tsx
Comment (satyam73) [inline src/components/ShimmerEffect/RedirectShimmer.tsx:1]: NIT: export default function RedirectShimmer(){
return <>your component code</>
}

File: __tests__/components/Redirect/ErrorPage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import ErrorPage from '@/components/Redirect/ErrorPage';
+
+describe('ErrorPage', () => {
+    it('should render correctly', () => {
+        render(<ErrorPage />);
+        expect(screen.getByText('Oops!')).toBeInTheDocument();
+        expect(screen.getByText('Something went wrong. Please try again.')).toBeInTheDocument();
+        expect(screen.getByText('Reload')).toBeInTheDocument();
+    });
```
Comment (satyam73) [inline __tests__/components/Redirect/ErrorPage.test.tsx:11]:
```suggestion
const errorText = screen.getByText('Oops!');
const somethingWentWrongText = screen.getByText('Something went wrong. Please try again.');
const reloadButton = screen.getByText('Reload');

        expect(errorText).toBeInTheDocument();
        expect(somethingWentWrongText ).toBeInTheDocument();
        expect(reloadButton ).toBeInTheDocument();
    });
```

**Review by satyam73 — CHANGES_REQUESTED** [review]

File: src/components/ShimmerEffect/RedirectShimmer.tsx
Comment (sahsisunny) [reply src/components/ShimmerEffect/RedirectShimmer.tsx:1]: why?

Comment (satyam73) [reply src/components/ShimmerEffect/RedirectShimmer.tsx:1]: It reduces the one line only being used for export although this is NIT so you can skip also, not necessary

File: src/pages/[redirect]/index.tsx (hunk 1 above)
Comment (satyam73) [reply src/pages/[redirect]/index.tsx:9]: Okay but you can customize I think in the package, usually I follow this order

- External Libraries 
- Internal Libraries (services, utils, custom hooks)
- Components
- Types
- Images
- CSS

File: src/components/ShimmerEffect/RedirectShimmer.tsx
Comment (satyam73) [reply src/components/ShimmerEffect/RedirectShimmer.tsx:1]: Readability SIr👀

File: src/pages/[redirect]/index.tsx (hunk 1 above)
Comment (sahsisunny) [reply src/pages/[redirect]/index.tsx:9]: If we change the rule too many files will been change  we need to create the separate issue ticket for this issue

File: src/components/ShimmerEffect/RedirectShimmer.tsx
Comment (sahsisunny) [reply src/components/ShimmerEffect/RedirectShimmer.tsx:1]: In each file we are exporting in the same way so we have to keep it like this also.

File: src/pages/[redirect]/index.tsx (hunk 1 above)
Comment (sahsisunny) [reply src/pages/[redirect]/index.tsx:9]: Btw there are too many imports here so we keep simple like third party library, local imports and styles and images

**Pull request merged**